File: go/3218. Minimum Cost for Cutting Cake I.go
func minimumCost(m int, n int, horizontalCut []int, verticalCut []int) int {
	answer, hc, vc := 0, 1, 1
	sort.Ints(horizontalCut)
	sort.Ints(verticalCut)
	i, j := len(horizontalCut)-1, len(verticalCut)-1
	for i >= 0 || j >= 0 {
		if j >= 0 && (i < 0 || horizontalCut[i] < verticalCut[j]) {
			answer += verticalCut[j] * hc
			vc++
			j--
		} else {
			answer += horizontalCut[i] * vc
			hc++
			i--
		}
	}
	return answer
}

File: java/3218. Minimum Cost for Cutting Cake I.java
class Solution {
    public int minimumCost(int m, int n, int[] horizontalCut, int[] verticalCut) {
        int hc = 1;
        int vc = 1;
        Arrays.sort(horizontalCut);
        Arrays.sort(verticalCut);
        int i = horizontalCut.length-1;
        int j = verticalCut.length-1;
        int answer = 0;
        while (i >= 0 || j >= 0) {
            if (j >= 0 && (i < 0 || horizontalCut[i] < verticalCut[j])) {
                vc++;
                answer += hc * verticalCut[j--];
            } else {
                hc++;
                answer += vc * horizontalCut[i--];
            }
        }
        return answer;
    }
}